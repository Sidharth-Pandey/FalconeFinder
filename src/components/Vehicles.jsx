import React from 'react';


const  Vehicles = ({vehicles, groupId, onVehicleChange})=>{
  const radioElememt = vehicles.map((vehicle)=>
  <div key = {vehicle.name} className = { (vehicle.total_no) ? '': 'disabled'}>
    <input type="radio" name = {groupId} id ={groupId} value={vehicle.name} onClick={onVehicleChange}
      disabled={!vehicle.total_no}/>
    <label>{vehicle.name}({vehicle.total_no})</label>
  </div>
);
return (
  <div className = "vehicleList">
    {radioElememt}
  </div>
);

}

export default Vehicles
