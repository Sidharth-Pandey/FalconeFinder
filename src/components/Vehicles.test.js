import React from 'react';
import ReactDOM from 'react-dom';
import Vehicles from './Vehicles';
import {shallow} from 'enzyme';


describe('<Vehicles/> test cases',()=>{
  const props = {
    vehicles:   [{"name":"Space pod","total_no":2,"max_distance":200,"speed":2},{"name":"Space rocket","total_no":1,"max_distance":300,"speed":4},{"name":"Space shuttle","total_no":1,"max_distance":400,"speed":5},{"name":"Space ship","total_no":2,"max_distance":600,"speed":10}],
    onVehicleChange : function(){},
    groupId : 0
  };
  const vehicles = shallow(<Vehicles {...props}/>);
  it('should match the snapshot', () => {
    expect(vehicles.html()).toMatchSnapshot();
  });

  it('it should have all the options provided by props', () => {
    expect(vehicles.find('input[type="radio"]').length).toEqual(4);
  });

  it('it should have proper planets value', () => {
    expect(vehicles.find('input[type="radio"]').at(0).props().value).toEqual('Space pod');
    expect(vehicles.find('input[type="radio"]').at(1).props().value).toEqual('Space rocket');
  });

  it('Radio button should be enable by default ', () => {
    expect(vehicles.find('input[type="radio"]').at(0).props().disabled).toBeFalsy();
    expect(vehicles.find('input[type="radio"]').at(1).props().disabled).toBeFalsy();
    expect(vehicles.find('input[type="radio"]').at(2).props().disabled).toBeFalsy();
    expect(vehicles.find('input[type="radio"]').at(3).props().disabled).toBeFalsy();
  });
})
