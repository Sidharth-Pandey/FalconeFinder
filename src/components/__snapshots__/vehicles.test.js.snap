// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vehicles/> test cases should match the snapshot 1`] = `"<div class=\\"vehicleList\\"><div class=\\"\\"><input type=\\"radio\\" name=\\"0\\" id=\\"0\\" value=\\"Space pod\\"/><label>Space pod(2)</label></div><div class=\\"\\"><input type=\\"radio\\" name=\\"0\\" id=\\"0\\" value=\\"Space rocket\\"/><label>Space rocket(1)</label></div><div class=\\"\\"><input type=\\"radio\\" name=\\"0\\" id=\\"0\\" value=\\"Space shuttle\\"/><label>Space shuttle(1)</label></div><div class=\\"\\"><input type=\\"radio\\" name=\\"0\\" id=\\"0\\" value=\\"Space ship\\"/><label>Space ship(2)</label></div></div>"`;
